packages/block-library/src/navigation/index.php

```diff
@@ -61,9 +71,9 @@ private static function is_responsive( $attributes ) {
 	 * @return bool Returns whether or not a navigation has a submenu.
 	 */
 	private static function has_submenus( $inner_blocks ) {
-		foreach ( $inner_blocks as $inner_block ) {
-			$inner_block_content = $inner_block->render();
-			$p                   = new WP_HTML_Tag_Processor( $inner_block_content );
+		$inner_blocks_content = static::get_inner_blocks_content( $inner_blocks );
```

Comment:
What we need to do is making suer inner blocks only render once for the navigation block. These are not the changes we need, but it helps demonstrate the impact.

Options:

1. Refactor the code for the inner blocks to be rendered once at WP_Navigation_Block_Renderer::render`.
2. Cache the inner blocks rendering.

I'd very much prefer 1, given this code is new and so we can make any modifications to it. Once it's in core we won't.